include/envoy/network/listen_socket.h

```diff
@@ -170,6 +170,16 @@ class ConnectionSocket : public virtual Socket {
    * @return requested server name (e.g. SNI in TLS), if any.
    */
   virtual absl::string_view requestedServerName() const PURE;
+
+  /**
+   * Set requested ECDSA cipher suite(s).
+   */
+  virtual void setEcdsaCipherSuites(const std::vector<uint16_t>& cipher_suites) PURE;
```

Comment:
If people have a legitimate wish to use the ECDSA ciphers, why not allow them to do this? If they don't have a need, they won't use them :)

Reply:
My point is that we should revert this whole PR once we get `SSL_IDENTITY`.
 
1. ECDSA ciphers are extracted here instead of inside TLS TransportSocket only because we already have TLS Inspector that extracts some information from `ClientHello` messages, but there is no use for this information outside of the TLS TransportSocket itself.

2. Contrary to `requestedServerName` and `requestedApplicationProtocols`, `ecdsaCipherSuites` are very TLS-specific information (and TLS Inspector is really a companion to TLS TransportSocket).

3. We don't extract other information from `ClientHello` messages (like: TLS version, cipher suites, whether the session is being resumed or not, etc.) and make them generally available, so I don't see a reason why we should expose supported ECDSA ciphers.